packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -304,6 +295,53 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
     // run all of our updates in parallel
     return Promise.all(operations);
   }
+
+  /**
+   * After a Lambda Function is updated, it cannot be updated again until the
+   * `State=Active` and the `LastUpdateStatus=Successful`.
+   *
+   * Depending on the configuration of the Lambda Function this could happen relatively quickly
+   * or very slowly. For example, Zip based functions _not_ in a VPC can take ~1 second whereas VPC
+   * or Container functions can take ~25 seconds (and 'idle' VPC functions can take minutes).
+   */
+  private async waitForLastUpdateStatusComplete(currentFunctionConfiguration: AWS.Lambda.FunctionConfiguration, lambda: AWS.Lambda): Promise<void> {
+    // if the function is deployed in a VPC or if it is a container image function
+    // then the update will take much longer and we can wait longer between checks
+    // otherwise, the update will be quick, so a 1-second delay is fine
+    const delaySeconds = !(!currentFunctionConfiguration.VpcConfig?.VpcId || currentFunctionConfiguration.VpcConfig.VpcId === '') ||
```

Comment:
I feel like the current formulation of this condition is overly complex.

Pretty sure we can simplify it to just:

```suggestion
    const delaySeconds = currentFunctionConfiguration.VpcConfig?.VpcId ||
```

(Remember that the empty string is falsy in JavaScript)